### ISSUE TITLE
[v2] get users memory error

### ISSUE THREAD

**Issue by douglsmith** [WP-API/WP-API#1446]
Most of my testing is working fine but I'm getting out of memory errors when trying to get users. I do not have this problem with the v1 API.

I'm using the REST API Console plugin on WordPress 4.2.3 with the beta v2 API plugin from the WordPress plugin repository. I have turned off all plugins other than the REST API and Console.

```
GET /wp/v2/users
200 – SyntaxError: JSON Parse error: Unrecognized token '<'
"<br /> <b>Fatal error</b>: Allowed memory size of 134217728 bytes exhausted (tried to allocate 19 bytes) in <b>/home/mysite/public_html/wp-includes/wp-db.php</b> on line <b>1379</b><br /> "
```

(I assume the syntax error part is just the console plugin since I don't see anything like that using the API through an outside client.)

I have a lot of users so I tried to limit it to 10 responses in case it was trying to bring back too many. No change.

```
GET /wp/v2/users?filter[number]=10
200 – SyntaxError: JSON Parse error: Unrecognized token '<'
"<br /> <b>Fatal error</b>: Allowed memory size of 134217728 bytes exhausted (tried to allocate 22 bytes) in <b>/home/mysite/public_html/wp-includes/wp-db.php</b> on line <b>1379</b><br /> "
```

`GET /wp/v2/users/me` or a specific ID works fine to bring back a singe user's data.

I set up a fresh site, installed everything new, and tried it there. It worked without problem. That makes me wonder if it's caused by the large number of users (over 50,000 from bbPress and WooCommerce). 

I'd be happy to do further troubleshooting and report back but I need some suggestions on how to best proceed next. Thanks.

**Comment by douglsmith** [comment]
I did some more testing and bumped my WordPress memory allocation up to 4096M. It was finally able to complete and return results. However, it took over two minutes to return only 10 users.

**Comment by rachelbaker** [comment]
@douglsmith Try `wp-json/wp/v2/users?per_page=1`

**Comment by douglsmith** [comment]
Thanks @rachelbaker. I was still getting memory errors with `wp-json/wp/v2/users?per_page=1`. I did more testing and I think the out of memory was related to transients bloat in my options table. Once I got that cleaned up I've been able to query users just fine. So I'm now guessing there's nothing that was the fault of the REST API unless there's some way to be more resilient in that circumstance.
